docs/calendar.md

```diff
@@ -347,6 +347,39 @@ Temporal.Calendar.from('iso8601').fields(['month', 'day']);
 ```
 <!-- prettier-ignore-end -->
 
+### calendar.**mergeFields**(_fields_: object, _additionalFields_: object) : object
+
+**Parameters:**
+
+- `fields` (object): A plain object with properties representing calendar units.
+- `additionalFields` (object): Another plain object with properties representing calendar units.
+
+**Returns:** a new object with properties from both `fields` and `additionalFields`.
+
+This method does not need to be called directly except in specialized code.
+It is called indirectly when using the `with()` methods of `Temporal.PlainDateTime`, `Temporal.PlainDate`, `Temporal.PlainMonthDay`, `Temporal.PlainYearMonth`, and `Temporal.ZonedDateTime`.
+
+Custom calendars should override this method if they allow a calendar unit to be specified in more than one way.
+(For example, the Gregorian calendar allows years to be specified either by a `year` property or a combination of `era` and `eraYear`.)
+The overridden implementation should return an object with the properties from the original `fields` object and `additionalFields` copied onto it, and possibly properties removed.
```

Comment:
Nit: properties are not removed from new object; something like "with some or all of the properties from..." might work better